pilot/pkg/model/gateway.go

```diff
@@ -173,6 +173,33 @@ func mergeGateways(gateways []gatewayWithInstances, proxy *Proxy, ps *PushContex
 	autoPassthrough := false
 
 	log.Debugf("mergeGateways: merging %d gateways", len(gateways))
+	// Check if the identity of the proxy matches the identity of a pod/workload of the gateway
+	identityVerified := false
+	checkedHosts := sets.New[NamespacedHostname]()
+	if proxy.VerifiedIdentity != nil {
+		spiffeID := proxy.VerifiedIdentity.String()
+	outer:
+		for _, gwAndInstance := range gateways {
```

Comment:
Is there a better way to verify this? I looked far and wide in push context and couldn't find a good way to match the verifiedIdentity with the gateway.

Reply:
Also, have a feature flag to restore old behavior?

Reply:
john and I backchanneled to speed things up and hopefully get this merged for 1.27.2. TLDR is technically, a gwapi `Gateway` with a `Hostname` address could be anything and there is no great way of knowing which service accounts are going will make istiod queries, since we don't manage the gateway instances directly, so we only check the service account namespace in these cases.